File: MCRunner/Orders/StopLimitOrder.cs
using PowerLanguage;
using System;

namespace MCRunner.Orders
{
    public class StopLimitOrder : BaseOrder, IOrderStopLimit
    {
        public StopLimitOrder(SOrderParameters orderParams, bool openNext = false)
            : base(orderParams, OrderCategory.StopLimit, openNext)
        {
        }

        public void Send(double stopPrice, double limitPrice)
        {
            if (stopPrice <= 0)
            {
                throw new ArgumentOutOfRangeException("stopPrice must be larger than 0");
            }

            if (limitPrice <= 0)
            {
                throw new ArgumentOutOfRangeException("limitPrice must be larger than 0");
            }

            TriggerOrderSent(new OrderInfo()
            {
                ConditionPrice = stopPrice,
                Order = this,
                OrderAction = OrderParams.Action,
                OrderExit = OrderParams.ExitTypeInfo,
                Price = limitPrice,
                Size = OrderParams.Lots.GetSize()
            });
        }

        public void Send(double stopPrice, double limitPrice, int numLots)
        {
            if (stopPrice <= 0)
            {
                throw new ArgumentOutOfRangeException("stopPrice must be larger than 0");
            }

            if (limitPrice <= 0)
            {
                throw new ArgumentOutOfRangeException("limitPrice must be larger than 0");
            }

            if (numLots < 0)
            {
                throw new ArgumentOutOfRangeException("numLots must be larger than 0");
            }
            else if (numLots == 0)
            {
                Send(stopPrice, limitPrice);
            }
            else
            {
                TriggerOrderSent(new OrderInfo()
                {
                    ConditionPrice = stopPrice,
                    Order = this,
                    OrderAction = OrderParams.Action,
                    OrderExit = OrderParams.ExitTypeInfo,
                    Price = limitPrice,
                    Size = OrderParams.Lots.GetSize(numLots)
                });
            }
        }

        public void Send(string new_name, double stopPrice, double limitPrice)
        {
            throw new NotImplementedException();
        }

        public void Send(string new_name, double stopPrice, double limitPrice, int numLots)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(double stopPrice, double limitPrice, string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(double stopPrice, double limitPrice, int numLots, string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(string new_name, double stopPrice, double limitPrice, string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(string new_name, double stopPrice, double limitPrice, int numLots, string fromName)
        {
            throw new NotImplementedException();
        }
    }
}

File: python/mcrunner/orders/stop_limit_order.py
from .enums import SOrderParameters, OrderCategory
from .order_info import OrderInfo
from .base_order import BaseOrder

class StopLimitOrder(BaseOrder):
    def __init__(self, order_params: SOrderParameters, open_next: bool = False):
        super().__init__(order_params, OrderCategory.StopLimit, open_next)

    def send(self, price: float, limit_price: float, num_lots: int = 0):
        if price <= 0 or limit_price <= 0:
            raise ValueError("price must be > 0")
        if num_lots < 0:
            raise ValueError("numLots must be >= 0")
        size = self.OrderParams.contracts.contract if not self.OrderParams.contracts.is_user_specified else num_lots or self.OrderParams.contracts.contract
        info = OrderInfo(
            ConditionPrice=price,
            Price=limit_price,
            Order=self,
            OrderAction=self.OrderParams.action,
            OrderExit=self.OrderParams.exit_type_info,
            Size=size,
        )
        self._trigger(info)
